java-squid/src/main/java/org/sonar/java/JavaClasspath.java

```diff
@@ -19,81 +19,50 @@
  */
 package org.sonar.java;
 
-import com.google.common.base.Splitter;
 import com.google.common.collect.Lists;
-import org.apache.commons.io.FileUtils;
-import org.apache.commons.io.FilenameUtils;
-import org.apache.commons.io.IOCase;
-import org.apache.commons.io.filefilter.AndFileFilter;
-import org.apache.commons.io.filefilter.DirectoryFileFilter;
-import org.apache.commons.io.filefilter.FalseFileFilter;
-import org.apache.commons.io.filefilter.IOFileFilter;
-import org.apache.commons.io.filefilter.OrFileFilter;
-import org.apache.commons.io.filefilter.TrueFileFilter;
-import org.apache.commons.lang.StringUtils;
 import org.apache.maven.artifact.DependencyResolutionRequiredException;
 import org.apache.maven.project.MavenProject;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.sonar.api.BatchExtension;
 import org.sonar.api.batch.fs.FilePredicates;
 import org.sonar.api.batch.fs.FileSystem;
 import org.sonar.api.batch.fs.InputFile;
 import org.sonar.api.config.Settings;
 import org.sonar.api.resources.Project;
 import org.sonar.api.utils.SonarException;
-import org.sonar.api.utils.WildcardPattern;
 
 import javax.annotation.Nullable;
 import java.io.File;
-import java.io.FileFilter;
-import java.util.Collection;
-import java.util.Iterator;
 import java.util.List;
 
-import static org.apache.commons.io.filefilter.FileFilterUtils.suffixFileFilter;
+public class JavaClasspath extends AbstractJavaClasspath {
 
-public class JavaClasspath implements BatchExtension {
 
-  private static final char SEPARATOR = ',';
-  private static final Logger LOG = LoggerFactory.getLogger(JavaClasspath.class);
-  private final Project project;
-  private final Settings settings;
-  private final FileSystem fs;
   @Nullable
   private final MavenProject pom;
-
-  private List<File> binaries;
-  private List<File> elements;
-  private boolean validateLibraries;
-  private boolean hasJavaSources;
-  private boolean initalized;
+  private static final Logger LOG = LoggerFactory.getLogger(JavaClasspath.class);
 
   public JavaClasspath(Project project, Settings settings, FileSystem fs) {
     this(project, settings, fs, null);
   }
 
   public JavaClasspath(Project project, Settings settings, FileSystem fs, @Nullable MavenProject pom) {
-    this.project = project;
-    this.settings = settings;
-    this.fs = fs;
+    super(project, settings, fs);
     this.pom = pom;
-    initalized = false;
   }
 
-  private void init() {
+  protected void init() {
     if (!initalized) {
       initalized = true;
       validateLibraries = project.getModules().isEmpty();
       FilePredicates predicates = fs.predicates();
       hasJavaSources = fs.hasFiles(predicates.and(predicates.hasLanguage("java"), predicates.hasType(InputFile.Type.MAIN)));
```

Comment:
`hasJavaSources` is set in both `JavaClasspath` and `JavaTestClasspath`, with the only difference being the  `InputFile` type (MAIN or TEST). A method using only the type as parameter can probably be put directly in the   `AbstractJavaClasspath` class.